tinygrad/lazy.py

```diff
@@ -257,8 +257,14 @@ def splitted_shape(dim_aft_div): return self.shape[:dim_to_split] + (self.shape[
   # *** movement ops ***
 
   def _movement_op(self, st: ShapeTracker, op: MovementOps, arg: Union[Tuple[sint, ...], Tuple[Tuple[sint, sint], ...]]) -> LazyBuffer:
-    if SHUFFLE_MOVEMENT_OPS and self.optype == BinaryOps and not self.realized and (op in {MovementOps.SHRINK, MovementOps.STRIDE, MovementOps.PERMUTE} or (op == MovementOps.RESHAPE and self.op.op in UnaryOps)) and not self.children:
-      return self.op.replace_with_movement_ops([(op, arg)])
+    if SHUFFLE_MOVEMENT_OPS and not self.realized and self.optype == BinaryOps and not self.children:
+      root_bufs = map(functools.partial(get_movementroot, allow_contiguous = True), self.op.buffers)
```

Comment:
`.base` might give you what you want here

Reply:
done thanks!